beater/server_test.go

```diff
@@ -160,16 +163,29 @@ func TestServerNoContentType(t *testing.T) {
 	assert.Equal(t, http.StatusBadRequest, rr.Code, rr.Body.String())
 }
 
+func TestServerSecureOkWithPasswordKey(t *testing.T) {
```

Comment:
This is good!
I miss 1 more test that "breaks" if the passphrase generated and the one passed in the server config do not match.
Note: in that scenario the server won't be able to start, so you need to assert that a timeout happens instead, the one in `waitForServer`. That is hardcoded to 10 seconds, so you will also need to modify that function to accept a `timeout` param and pass a lower value for that test because 10 secs every time it runs would be very slow.
Does this makes sense to you?